cobbler/remote.py

```diff
@@ -1514,8 +1515,9 @@ def remove_item(
         self._log(
             f"remove_item ({what}, recursive={recursive})", name=name, token=token
         )
-        obj = self.api.get_item(what, name)
-        if obj is None:
+        try:
+            obj = self.__get_object(self.get_item_handle(what, name))
+        except ValueError:
             return False
         self.check_access(token, f"remove_{what}", obj.name)
         self.api.remove_item(
```

Comment:
If you do this then you need to handle the case that the item is not added to the API and as such only delete it from the local cache. That operation is complex enough to warrant tests.